### PR TITLE
Feature/video player

### PR DESCRIPTION
**Pull request by yalania**
- Added "resource" video, is not real resource as it will stream directly from the file. For now the same video can't be play in different frames
- I fixed some warnings on the code that were annoying me
- I have do the extract data more generic, because we were creating a class for resources that doesn't need it

To use:
Move the .mp4 you want to see to assets folder
Create component video player
Enable video and drag and drop themp4 you want to see
Click play (in video and game)

Note: Make sure you have an audio listener in order to hear the audio.

### CODE REVIEW

**Review by mariofv — COMMENTED** [review]
Work as expected. Few questions:
- Why is the functionality inside `ComponentImage`, and not in a new Component?
- Would it be possible to also include Audio without puting a SoundBank?

**Comment by yalania** [comment]
I create a new computer ment at the beginning but I was just duplicating code from component image, at then a video is just a set of images changes. 

Opencv, the library I used is only for images it doesn't manage sound, so is not possible to do it.

**Comment by mariofv** [comment]
> I create a new computer ment at the beginning but I was just duplicating code from component image, at then a video is just a set of images changes.

So then, could it be possible to add the sound resource of the video inside the component? This way everything would be more organized, and then maybe I would have sense to create a new component with the video resource + sound

**Comment by yalania** [comment]
You don't let me be lazy :( Ok I will do it

**Comment by mariofv** [comment]
Code looks ok, but I cannot try this until you merge develop :P

File: Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp [hunk 1]
```diff
@@ -690,6 +691,57 @@ void PanelComponent::ShowComponentImageWindow(ComponentImage* component_image)
 	}
 }
 
+void PanelComponent::ShowComponentVideoPlayerWindow(ComponentVideoPlayer* video_player)
+{
+
```
Comment (mariofv) [inline Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp]: You forgot here to show the `CommonComponentWindow`, and this causes the Component Ui to show without the "Header":
image

**Review by mariofv — CHANGES_REQUESTED** [review]
SetAspectRatio is not working for both `ComponentImage` and `ComponentVideo`, also look at the coment. Rest is fine, good job!

**Comment by yalania** [comment]
Aspect ratio is not working in master either

**Review by mariofv — APPROVED** [review]

**Review by JorxSS — APPROVED** [review]
Sorry for the delay, but hopefully you will pardon me because ...
image

Good Job Anabel!

**Pull request merged**